bundles/org.openhab.binding.sonyprojector/src/main/java/org/openhab/binding/sonyprojector/internal/communication/SonyProjectorConnector.java

```diff
@@ -881,6 +882,64 @@ public void disableXvColor() throws SonyProjectorException {
         setSetting(SonyProjectorItem.XVCOLOR, XVCOLOR_OFF);
     }
 
+    /**
+     * Send an IR command to the projector
+     *
+     * @param command the IR command
+     *
+     * @throws SonyProjectorException in case this IR command is not available or any other problem
+     */
+    public void sendIrCommand(String command) throws SonyProjectorException {
+        byte @Nullable [] irCode = null;
+        boolean found = true;
+        if (command.startsWith("INPUT_")) {
+            try {
+                irCode = model.getInputFromName(command.substring(6)).getIrCode();
+            } catch (SonyProjectorException e) {
+                found = false;
+            }
+        } else if (command.startsWith("PRESET_")) {
+            try {
+                irCode = model.getCalibrPresetFromName(command.substring(7)).getIrCode();
+            } catch (SonyProjectorException e) {
+                found = false;
+            }
+        } else if (command.startsWith("ASPECT_")) {
+            try {
+                irCode = model.getAspectFromName(command.substring(7)).getIrCode();
+            } catch (SonyProjectorException e) {
+                found = false;
+            }
+        } else {
+            try {
+                irCode = SonyProjectorItem.getFromValue(command).getIrCode();
+            } catch (SonyProjectorException e) {
+                found = false;
+            }
+        }
+        if (!found) {
+            // Check if the command is a direct IR code in hexadicimal
+            // Must be 4 characters starting by either 17 or 19 or 1B
+            String cmd = command.trim();
+            if (cmd.length() != 4) {
+                throw new SonyProjectorException("Invalid IR code: " + command);
+            }
+            int hex;
+            try {
+                hex = Integer.parseInt(cmd, 16);
+            } catch (NumberFormatException e) {
+                throw new SonyProjectorException("Invalid IR code: " + command);
```

Comment:
You could consider providing the cause as well:
```suggestion
                throw new SonyProjectorException("Invalid IR code: " + command, e);
```